tools/swift-serialize-diagnostics/swift-serialize-diagnostics.cpp

```diff
@@ -0,0 +1,80 @@
+//===--- swift-serialize-diagnostics.cpp ----------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Convert localization YAML files to a srialized format.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/AST/LocalizationFormat.h"
+#include "swift/Basic/LLVMInitialize.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Bitstream/BitstreamReader.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/YAMLParser.h"
+#include "llvm/Support/YAMLTraits.h"
+
+#include <iostream>
+
+using namespace swift;
+using namespace swift::diag;
+
+namespace {
+static constexpr const char *const diagnosticID[] = {
+#define DIAG(KIND, ID, Options, Text, Signature) #ID,
+#include "swift/AST/DiagnosticsAll.def"
+};
+} // namespace
+
+namespace options {
+
+static llvm::cl::OptionCategory Category("swift-serialize-diagnostics Options");
+
+static llvm::cl::opt<std::string>
+    InputFilePath("input-file-path", llvm::cl::desc("Path of the input file"),
+                  llvm::cl::cat(Category));
+
+static llvm::cl::opt<std::string>
+    OutputDirectory("output-directory",
+                    llvm::cl::desc("Directory for the output file"),
+                    llvm::cl::cat(Category));
+
+} // namespace options
+
+int main(int argc, char *argv[]) {
+  PROGRAM_START(argc, argv);
+  INITIALIZE_LLVM();
+
+  llvm::cl::HideUnrelatedOptions(options::Category);
+  llvm::cl::ParseCommandLineOptions(argc, argv,
+                                    "Swift Serialize Diagnostics Tool\n");
+
+  if (!llvm::sys::fs::exists(options::InputFilePath)) {
+    llvm::errs() << "YAML file not found\n";
+    return 1;
+  }
+
+  YAMLLocalizationProducer YAMLDiagnostics(options::InputFilePath);
+
+  auto localeCode = llvm::sys::path::filename(options::InputFilePath);
+  llvm::SmallString<128> SerializedFilePath(options::OutputDirectory);
+  llvm::sys::path::append(SerializedFilePath, localeCode);
+  llvm::sys::path::replace_extension(SerializedFilePath, ".db");
+
+  SerializedLocalizationWriter Serializer;
+  for (unsigned id = 0; id < YAMLDiagnostics.diagnostics.size(); ++id)
+    Serializer.insert(diagnosticID[id], YAMLDiagnostics.diagnostics[id]);
+
+  Serializer.emit(SerializedFilePath.str());
```

Comment:
Let's check whether `emit` has successful and if not - write an error out and return `-1;`